crates/holochain/src/core/workflow/app_validation_workflow.rs

```diff
@@ -650,58 +646,42 @@ async fn run_validation_callback(
         ValidateResult::Invalid(reason) => Ok(Outcome::Rejected(reason)),
         ValidateResult::UnresolvedDependencies(UnresolvedDependencies::Hashes(hashes)) => {
+            tracing::debug!(
+                ?hashes,
+                "Op validation returned unresolved dependencies -  AgentActivity"
+            );
             // fetch all missing hashes in the background without awaiting them
             let cascade_workspace = workspace.clone();
             let cascade =
                 CascadeImpl::from_workspace_and_network(&cascade_workspace, network.clone());
 
-            // add hashes missing to validate an op to hash map
-            hashes.iter().for_each(|hash| {
-                validation_dependencies
-                    .lock()
-                    .insert_hash_missing_for_op(dht_op_hash.clone(), hash.clone());
-            });
-
-            // build a collection of futures to fetch the individual missing
-            // hashes
-            let validation_deps = validation_dependencies.clone();
-            let dht_op_hash = dht_op_hash.clone();
-            let fetches = hashes
-                .clone()
-                .into_iter()
-                .filter(move |hash| {
-                    // keep track of which dependencies are being fetched to
-                    // prevent multiple fetches of the same hash
-                    let is_new_dependency = validation_deps
-                        .lock()
-                        .insert_missing_hash(hash.clone())
-                        .is_none();
-                    is_new_dependency
-                })
-                .map(move |hash| {
-                    let cascade = cascade.clone();
-                    let validation_dependencies = validation_dependencies.clone();
-                    let dht_op_hash = dht_op_hash.clone();
-                    async move {
-                        let result = cascade
-                            .fetch_record(hash.clone(), NetworkGetOptions::must_get_options())
-                            .await;
-                        if let Err(err) = result {
-                            tracing::warn!("error fetching dependent hash {hash:?}: {err}");
-                        }
-                        // Dependency has been fetched and added to the cache
-                        // or an error occurred along the way.
-                        // In case of an error the hash is still removed from
-                        // the collection so that it will be tried again to be
-                        // fetched in the subsequent workflow run.
-                        validation_dependencies.lock().remove_missing_hash(&hash);
-                        // Secondly remove the just fetched hash from the set
-                        // of missing hashes for the op
-                        validation_dependencies
-                            .lock()
-                            .remove_hash_missing_for_op(dht_op_hash, &hash);
```

Comment:
... and therefore never removed from the list of ops missing hashes, which leads to the filtering out above.